server/controllers/Article.js

```diff
@@ -0,0 +1,217 @@
+  
+    const authorId = req.body.decoded.id;
+  
+  console.log(req.body.decoded.id);
+    Article.create({
+      slug, title, description, body, authorId,
+    })
+      .then((article) => {
+        res.status(201).json({
+          article: {
+            id: article.id,
+            title: article.title,
+            description: article.description,
+            body: article.body,
+            slug,
+          },
+          message: 'Article Sucessfully Created',
+        });
+      })
+      .catch(next);
+  };
+  
+  static getArticle(req, res, next) {
+    Article.find({ where: { slug: req.params.slug } })
+      .then((article) => {
+        if (article) {
+          User.find({ where: { id: article.authorId } })
+            .then((user) => {
+              res.status(201).json({
+                article: {
+                  slug: article.slug,
+                  title: article.title,
+                  description: article.description,
+                  body: article.body,
+                  author: user.username,
+                },
+                message: 'Article Found',
+              });
+            })
+            .catch(next);
+        } else {
+          res.status(404).json({
+            status: 'fail',
+            message: 'Article not found',
+          });
+        }
+      })
+      .catch(next);
+  };
+  
+  static updateArticle(req, res, next) {
+    const {
+      title, description, body,
+    } = req.body.article;
+  
+    Article.find({ where: { slug: req.params.slug } })
+      .then((article) => {
+        if (article) {
+          const updatedArticle = article;
+          updatedArticle.title = title !== undefined ? title : article.title;
+          updatedArticle.description = description !== undefined ? description : article.description;
+          updatedArticle.body = body !== undefined ? body : article.body;
+          updatedArticle.save();
+          User.find({ where: { id: article.authorId } })
+            .then((user) => {
+              res.status(200).json({
+                article: {
+                  slug: article.slug,
+                  title: article.title,
+                  description: article.description,
+                  body: article.body,
+                  author: user.username,
+                },
+                message: 'Article Updated',
+              });
+            })
+            .catch(next);
+        } else {
+          res.status(404).json({
+            status: 'fail',
+            message: 'Article not found',
+          });
+        }
+      })
+      .catch(next);
+  };
+  
+  static deleteArticle(req, res, next) {
+    Article.find({ where: { slug: req.params.slug } })
+      .then((article) => {
+        if (article) {
+          Article.destroy({ where: { slug: req.params.slug } })
+            .then((success) => {
+              if (success) {
+                res.status(200).json({
+                  article: {
```

Comment:
You are returning an article after deleting it. Please, can you explain what you are returning?